mh/pat.py

```diff
@@ -2451,8 +2457,12 @@ def handle_client(self):
                 # Send a ping with 30 seconds interval
                 current_time = time.time()
                 if current_time > timeout:
-                    self.sendReqLineCheck()
-                    timeout = current_time + 30
+                    if self.timeout_passed or self.server.debug_enabled():
+                        self.timeout_passed = False
+                        self.sendReqLineCheck()
+                        timeout = current_time + 30
+                    else:
+                        raise Exception("Client timed out.")
```

Comment:
The condition should be inverted, it will reduce the indentation level by one.
```python
if not self.server.debug_enabled() and not self.line_check:
    raise Exception("Client timed out.")
self.line_check = False
self.sendReqLineCheck()
timeout = current_time + 30
```

An assert can be used instead of rising an exception.

Reply:
Fixed